src/Symfony/Component/Validator/Tests/Constraints/NotJsonValidatorTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\NotJson;
+use Symfony\Component\Validator\Constraints\NotJsonValidator;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+class NotJsonValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator()
+    {
+        return new NotJsonValidator();
+    }
+
+    public function testJsonIsValid()
+    {
+        $this->validator->validate('{ "planet":"earth", "country": "Morocco", "city": "Rabat" }', new NotJson());
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getInvalidValues
+     */
+    public function testInvalidValues($value)
+    {
+        $constraint = new NotJson(array(
+            'message' => 'myMessageTest',
+        ));
+
+        $this->validator->validate($value, $constraint);
+
+        $this->buildViolation('myMessageTest')
+            ->setParameter('{{ value }}', '"'.$value.'"')
+            ->setCode(NotJson::NOT_JSON_ERROR)
+            ->assertRaised();
+    }
+
+    public function getInvalidValues()
+    {
+        return array(
+            array('{"foo": 3 "bar": 4}'),
+            array('{"foo": 3 ,"bar": 4'),
+            array('{foo": 3, "bar": 4}'),
+        );
+    }
```

Comment:
We should add a test to check, that for `""` and `null` the validator doesn't add violations.

Reply:
I've added the tests, and the code to support this two cases.